### ISSUE TITLE
[Bug Report] Console is appearing again on stash.exe launch as of 0.15.0

### ISSUE THREAD

**Issue by hphpanon** [stashapp/stash#2598]
**Describe the bug**
Release [0.13.0](https://github.com/stashapp/stash/releases/tag/v0.13.0) introduced the behavior that Stash would open solely in the systray and no longer open a console window. 0.15.0 has regressed that. Most likely related to https://github.com/stashapp/stash/pull/2543 (which directly manipulated behavior around Windows console stuff, but I'm no Go expert).

The stash icon appears in the systray, but I also get the terminal window, which I wasn't getting as of 0.13.0 and don't want anymore.

**To Reproduce**
1. Launch a 0.15.0 stash.exe on Windows via double-click in Windows Explorer

**Expected behavior**
Stash running with just the stash icon in the systray. No terminal window.

**Screenshots**
![image](https://user-images.githubusercontent.com/102124543/169097326-d912609c-1cc7-4c33-8b95-ec708112b5f9.png)

**Stash Version: (from Settings -> About):** v0.15.0 (build hash `b2ac0223`)

**Desktop (please complete the following information):**
 - OS: Windows 11 21H2 (22000.675)
 - Browser: Edge 101.0.1210.47

**Additional context**
As hopefully evident in the screenshot, I've changed the default console/terminal app to Windows Terminal instead of the legacy Command Prompt app. I can't imagine that's particularly relevant to the behavior in question, but figured I'd mention it just in case.

**Comment by hphpanon** [comment]
> As hopefully evident in the screenshot, I've changed the default console/terminal app to Windows Terminal instead of the legacy Command Prompt app. I can't imagine that's particularly relevant to the behavior in question, but figured I'd mention it just in case.

After looking through the #2543, it appears this is actually extremely relevant. If I set my default terminal app to the legacy command prompt app, it behaves as expected. However, previous versions of Stash did not have this limitation, so this is still a regression in my eyes.

**Comment by hphpanon** [comment]
It looks like launching Stash creates an OpenConsole.exe process. I don't know for certain that this is new behavior with 0.15.0, but given that it appears to be trying to open the old cmd.exe app that seems likely.
![image](https://user-images.githubusercontent.com/102124543/169099993-251c97a1-76bb-45c8-95cd-628406bb9438.png)

(The Windows Terminal process not shown is just the semi-standard Runtime Broker for Windows apps)

**Comment by MrX292** [comment]
does the console stay open? for me it closes after a couple of seconds and is only in the systray.

**Comment by hphpanon** [comment]
Yes, it stays open.

**Comment by hphpanon** [comment]
If i set my default terminal to the legacy Command Prompt it works as expected (shows up briefly and then disappears) but with it set to Windows Terminal, it stays open as long as stash is running. In Stash 0.13.x and 0.14.0, it was fully compatible with Windows Terminal being the default terminal app.

**Comment by WithoutPants** [comment]
This looks like the same bug as https://github.com/nzbget/nzbget/issues/792

There is also a workaround posted in that issue. Could you see if that resolves the issue for you?

**Comment by hphpanon** [comment]
Yup, that works. It looks like Microsoft is working on making Windows Terminal [support some of the functionality in question](https://github.com/microsoft/terminal/issues/12570), so it might Just Work™ in a future Terminal update (assuming there's nothing else that can be done on the stash side of things).

In any case, I can once again have stash happily run in the background. I'll leave it up to you whether this should be closed (since it has a valid workaround) or stay open. Thanks for finding that!

**Comment by djbtwothousandtwenty** [comment]
Bumped into this one starting somewhere last week as well, but in my case it's because of 2 things:

- I'm running a preview build of Windows 11 "next"
- Microsoft has changed the default terminal from command prompt to their new Terminal in the build i updated to last week: https://blogs.windows.com/windows-insider/2022/08/24/announcing-windows-11-insider-preview-build-25188/

This means this will likely happen to a lot more people as soon as this update reaches the mainstream windows 11 users.

**Comment by Phasetime** [comment]
Workaround for me is running stash as admin. It seems to not run in Windows Terminal then and works fine again, even with Windows Terminal being the default terminal.

**Comment by WeedLordVegeta420** [comment]
Using a shortcut that starts stash directly through conhost.exe as suggested in https://github.com/nzbget/nzbget/issues/792 worked for me.

**Comment by xx790** [comment]
Still an issue.
Made `wt` the default terminal and got this issue.
The mentioned workaround with `conhost` link does work.

Stash:
```
v0.21.0
Build hash 9180a68c
```

OS:
```
Edition  Windows 10 Pro
Version  21H2
OS build 19044.3086
```

Windows Terminal: 
```
1.17.11461.0
```

**Comment by jittarao** [comment]
> Using a shortcut that starts stash directly through conhost.exe as suggested in [nzbget/nzbget#792](https://github.com/nzbget/nzbget/issues/792) worked for me.

This worked for me. Thank you :)

**Comment by WithoutPants** [comment]
I'm going to close this as a Wontfix since there's no real good solution on the code side and there are ways to achieve this that don't involve code or build changes.

It is possible to compile stash as a windows application (as opposed to a console one) by adding `-H=windowsgui` to the `ldflags` ([source](https://stackoverflow.com/a/36728885/695786)). This worked in that you could double click the exe and no console window would appear. However, you cannot run the binary from the console and see the output, so it won't run as a console app at all. This is a non-starter for the default windows binary. We could consider building a separate windows gui specific binary, but I don't think it's worth doing so, and would probably cause confusion.

For users that want to hide the console, there are two methods ([source](https://stackoverflow.com/questions/75877438/how-to-hide-console-in-windows-11)):
- create a shortcut, setting the target to `conhost <path to stash>\stash-win.exe`.
- create a batch file to open stash using powershell. Example below:
```
@echo off
powershell.exe -Command "Start-Process 'stash-win.exe' -WindowStyle Hidden"
```
